### PR TITLE
CIV-14664 Service PR to update Judgments Tab in ExUI

### PR DESCRIPTION
**Pull request by sjamihmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-14664

### Change description ###
Service PR to update Judgments Tab in ExUI

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentOnlineMapper.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public JudgmentDetails addUpdateActiveJudgment(CaseData caseData) {
         isNonDivergent =  JudgmentsOnlineHelper.isNonDivergentForDJ(caseData);
         JudgmentDetails activeJudgment = super.addUpdateActiveJudgment(caseData);
         activeJudgment = super.updateDefendantDetails(activeJudgment, caseData);
-        return activeJudgment.toBuilder()
+        JudgmentDetails activeJudgmentDetails = activeJudgment.toBuilder()
```
Comment (sjamihmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentOnlineMapper.java:40]: Revert this change

File: src/test/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentsOnlineMapperTest.java [hunk 2]
```diff
@@ -67,7 +67,6 @@ void testIfDefaultJudgmentIsMarkedActive_1v1() {
         assertEquals("Mr. Sole Trader", activeJudgment.getDefendant1Name());
         assertNotNull(activeJudgment.getDefendant1Address());
         assertNotNull(activeJudgment.getDefendant1Dob());
-
```
Comment (sjamihmcts) [inline src/test/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentsOnlineMapperTest.java:70]: Remove extra lines

File: src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentOnlineMapper.java (hunk 1 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentOnlineMapper.java]: Done

File: src/test/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentsOnlineMapperTest.java (hunk 2 above)
Comment (hmcts-version1-ignacio) [reply src/test/java/uk/gov/hmcts/reform/civil/helpers/judgmentsonline/DefaultJudgmentsOnlineMapperTest.java:70]: I still see empty lines in this file.

**Review by hmcts-version1-ignacio — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java [hunk 3]
```diff
@@ -98,14 +98,17 @@ private CallbackResponse buildConfirmation(CallbackParams callbackParams) {
     private CallbackResponse saveJudgmentDetails(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
         List<String> errors = new ArrayList<>();
+
         if (caseData.getJoIsRegisteredWithRTL() == YesOrNo.YES) {
             caseData.setJoIssuedDate(caseData.getJoOrderMadeDate());
         }
         if (caseData.getActiveJudgment() != null) {
-            caseData.setActiveJudgment(editJudgmentOnlineMapper.addUpdateActiveJudgment(caseData));
+            editJudgmentOnlineMapper.moveToHistoricJudgment(caseData, false);
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java:106]: Why we need to move to historic judgment?

Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java:106]: As per Scenario 5 in acceptance criteria of CIV-14664, we need to archive the previous judgment and save the updated one as activeJudgment

Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java:106]: image

Comment (Azam-Hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java:106]: It shouldnt ...this doesnt make any sense. I think its added by mistake. Why we want to move to historic judgment when judgment is edited. It should be only moved to historic when its set A side.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DefaultJudgementSpecHandler.java [hunk 4]
```diff
@@ -433,10 +442,7 @@ private CallbackResponse generateClaimForm(CallbackParams callbackParams) {
             nextState = CaseState.PROCEEDS_IN_HERITAGE_SYSTEM.name();
             caseDataBuilder.businessProcess(BusinessProcess.ready(DEFAULT_JUDGEMENT_SPEC));
         }
-        if (featureToggleService.isJudgmentOnlineLive()) {
-            caseDataBuilder.activeJudgment(djOnlineMapper.addUpdateActiveJudgment(caseData));
-            caseDataBuilder.joIsLiveJudgmentExists(YesOrNo.YES);
-        }
```
Comment (sjamihmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DefaultJudgementSpecHandler.java:439]: Update to caseDataBuilder

Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/DefaultJudgementSpecHandler.java:439]: Done.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java [hunk 5]
```diff
@@ -98,14 +98,17 @@ private CallbackResponse buildConfirmation(CallbackParams callbackParams) {
     private CallbackResponse saveJudgmentDetails(CallbackParams callbackParams) {
         CaseData caseData = callbackParams.getCaseData();
         List<String> errors = new ArrayList<>();
+
         if (caseData.getJoIsRegisteredWithRTL() == YesOrNo.YES) {
             caseData.setJoIssuedDate(caseData.getJoOrderMadeDate());
         }
         if (caseData.getActiveJudgment() != null) {
             caseData.setActiveJudgment(editJudgmentOnlineMapper.addUpdateActiveJudgment(caseData));
+            caseData.setJoRepaymentSummaryObject(JudgmentsOnlineHelper.calculateRepaymentBreakdownSummary(caseData.getActiveJudgment()));
```
Comment (hmcts-version1-ignacio) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java:107]: Should it be changed to caseDataBuilder here too?

Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/EditJudgmentCallbackHandler.java]: Done.

**Review by hmcts-version1-ignacio — DISMISSED** [review]

**Review by Omaira-Melo-Hmcts — DISMISSED** [review]

**Review by sherlynkhaw — APPROVED** [review]

**Review by kalachandrasekar1 — DISMISSED** [review]

**Review by Omaira-Melo-Hmcts — APPROVED** [review]

**Review by Azam-Hmcts — APPROVED** [review]
